### PR TITLE
feat(wasmbus): implement actor -> provider linking

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

Actor -> Provider linking via `wasmcloud:messaging` interface (interface itself only partially implemented)

## Related Issues

Closes https://github.com/wasmCloud/wasmCloud/issues/375

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Manual Verification

`cargo test`

### CODE REVIEW

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -196,50 +198,319 @@ struct ActorInstance {
...
+                let invocation =
+                    Invocation::new(&cluster_key, origin, target, interface_method, request)
+                        .map_err(|e| e.to_string())?;
+                let request = rmp_serde::to_vec_named(&invocation)
+                    .context("failed to encode invocation")
+                    .map_err(|e| e.to_string())?;
+                let res = nats
+                    .request(
+                        format!("wasmbus.rpc.default.{provider_id}.default"),
+                        request.into(),
```
Comment (brooksmtownsend) [inline crates/host/src/wasmbus/mod.rs:388]: #432 should come back and introduce the lattice prefix here

File: crates/host/src/wasmbus/mod.rs [hunk 2]
```diff
@@ -196,50 +198,319 @@ struct ActorInstance {
...
+        operation: String,
+        request: Vec<u8>,
+    ) -> anyhow::Result<Result<(), String>> {
+        let invocation = self.invocation(operation, request).await?;
+        let request =
+            rmp_serde::to_vec_named(&invocation).context("failed to encode invocation")?;
+        self.nats
+            .publish(
+                format!(
+                    "wasmbus.rpc.default.{}.default",
```
Comment (brooksmtownsend) [inline crates/host/src/wasmbus/mod.rs:417]: Same comment re: #432

File: crates/host/src/wasmbus/mod.rs [hunk 3]
```diff
@@ -196,50 +198,319 @@ struct ActorInstance {
...
+        response: &mut Vec<u8>,
+    ) -> anyhow::Result<Result<usize, String>> {
+        let invocation = self.invocation(operation, request).await?;
+        let request =
+            rmp_serde::to_vec_named(&invocation).context("failed to encode invocation")?;
+        let mut res = self
+            .nats
+            .request(
+                format!(
+                    "wasmbus.rpc.default.{}.default",
```
Comment (brooksmtownsend) [inline crates/host/src/wasmbus/mod.rs:440]: same comment re: #432

**Review by brooksmtownsend — APPROVED** [review]

File: crates/host/src/wasmbus/mod.rs [hunk 4]
```diff
@@ -196,50 +198,319 @@ struct ActorInstance {
...
+            pub subject: String,
+            #[serde(rename = "replyTo")]
             #[serde(skip_serializing_if = "Option::is_none")]
-            content_length: Option<u64>,
+            pub reply_to: Option<String>,
+            #[serde(with = "serde_bytes")]
+            pub body: Vec<u8>,
         }
+
+        const METHOD: &str = "wasmcloud:messaging/Messaging.Publish";
```
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply crates/host/src/wasmbus/mod.rs:497]: The "WIT way" is all-lowercase, but the legacy Smithy interfaces capitalize first character. Here we "translate" the WIT invocation into a Smithy one, so we can't change the format.

Going forward, we should be using the all-lowercase format, just like all other WASI interfaces.

**Pull request merged**